scripts/deploy-buildkite-darwin/darwin-configuration.nix

```diff
@@ -0,0 +1,124 @@
+  environment.systemPackages = with pkgs; [
+    nix-repl
+    nix
+    tmux
+    ncdu
+    git
+  ] ++ (if pkgs.stdenv.isDarwin then [
+    darwin.cctools
+  ] else []);
+
+  # Create /etc/bashrc that loads the nix-darwin environment.
+  programs.bash.enable = true;
+  # programs.zsh.enable = true;
+  # programs.fish.enable = true;
+
+  # Used for backwards compatibility, please read the changelog before changing.
+  # $ darwin-rebuild changelog
+  system.stateVersion = 2;
+
+  # You should generally set this to the total number of logical cores in your system.
+  # $ sysctl -n hw.ncpu
+  nix.maxJobs = 4;
+  nix.buildCores = 1;
+  nix.useSandbox = false;  # this seems to break things when enabled
+  nix.extraOptions = ''
+    gc-keep-derivations = true
+    gc-keep-outputs = true
+  '';
+
+  nix.binaryCachePublicKeys = [ "hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ=" ];
+  nix.binaryCaches = [ "https://hydra.iohk.io" ];
+  nix.trustedUsers = [ "@admin" ];
+
+  nix.nixPath = [
+    "nixpkgs=https://nixos.org/channels/nixpkgs-unstable/nixexprs.tar.xz"
+    "darwin=https://github.com/LnL7/nix-darwin/archive/master.tar.gz"
+    "darwin-config=/Users/admin/.nixpkgs/darwin-configuration.nix"
+    "iohk-ops=https://github.com/input-output-hk/iohk-ops/archive/master.tar.gz"
+  ];
+
+  ########################################################################
+
+  # try to ensure 25G of free space
+  nix.gc.automatic = true;
+  nix.gc.options = "--max-freed $((25 * 1024**3 - 1024 * $(df -P -k /nix/store | tail -n 1 | awk '{ print $4 }')))";
+
+  ########################################################################
+
+  services.nix-daemon.enable = true;
+
+  # Recreate /run/current-system symlink after boot.
+  services.activate-system.enable = true;
+
+  system.activationScripts.postActivation.text = ''
+    printf "disabling spotlight indexing... "
+    mdutil -i off -d / &> /dev/null
+    mdutil -E / &> /dev/null
+    echo "ok"
+  '';
+
+  ########################################################################
+
+  services.buildkite-agent = let
+    # something off the nixos-unstable branch
+    unstablePkgs = import (pkgs.fetchFromGitHub {
+      owner = "NixOS";
+      repo = "nixpkgs";
+      rev = "8b1cf100cd8badad6e1b6d4650b904b88aa870db";
+      sha256 = "1p0xxyz30bd2bg0wrfviqgsskz00w647h0l2vi33w90i42k8r3li";
+    }) { inherit config; };
+    # grabs a single program out of a system path
+    symlinkBin = prog: pkgs.runCommand (baseNameOf prog) {} ''
+      mkdir -p $out/bin
+      ln -s ${prog} $out/bin
+    '';
+  in {
+    enable = true;
+    package = unstablePkgs.buildkite-agent;
+    runtimePackages = with pkgs; [ bash nix git ] ++
+      map symlinkBin [ "/usr/sbin/pkgutil" ] ++
+      # These are required for daedalus build.
+      # fixme: update daedalus pipeline scripts so that it declares all of its dependencies
+      [ python27 gnumake curl nodejs-6_x nodePackages.node-gyp nodePackages.node-pre-gyp electron ];
```

Comment:
This shouldn't be here and it should belong to daedalus as noted - I think we should fix it before merging this.